### PR TITLE
Feat/ayt 3400/launching authoring by new configurable roles

### PR DESCRIPTION
**Pull request by bartlomiejmarszal**
How to reproduce:

Launch authoring tool using `devkit` [link](https://devkit.docker.localhost/platform/message/launch/lti-resource-link?registration=devkit__backoffice&user_type=list&user_list=r2d2&launch_url=http://backoffice.docker.localhost/taoLti/AuthoringTool/launch&claims=%7B%0D%0A%20%20%20%20%22https://purl.imsglobal.org/spec/lti/claim/roles%22:%20%5B%0D%0A%20%20%20%20%20%20%20%20%22http://purl.imsglobal.org/vocab/lis/v2/membership/ContentDeveloper%23ContentDeveloper%22,%0D%0A%20%20%20%20%20%20%20%20%22http://purl.imsglobal.org/vocab/lis/v2/membership/Administrator%23Developer%22%0D%0A%20%20%20%20%5D%0D%0A%7D)

Launch without allowed roles: 
- `http://purl.imsglobal.org/vocab/lis/v2/membership#Instructor`
- `http://purl.imsglobal.org/vocab/lis/v2/institution/person#Administrator`
- `http://purl.imsglobal.org/vocab/lis/v2/membership/ContentDeveloper#ContentDeveloper`
-  `http://purl.imsglobal.org/vocab/lis/v2/membership/Administrator#Developer`

https://github.com/oat-sa/extension-tao-lti/assets/16231681/0208fe07-0214-4eb6-9fc6-52627d7a4a0f

### CODE REVIEW

**Comment by andreluizmachado** [comment]
Hey, nice job! when you have time, could you please provide a demo????

**Review by andreluizmachado — COMMENTED** [review]
Hey, nice job! when you have time, could you please provide a demo????

**Comment by andreluizmachado** [comment]
Also, please, take a look at the ci jobs. Some of them are failing.
Last question: Can we open it on iframe?

**Review by Karol-Stelmaczonek — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Comment by Karol-Stelmaczonek** [comment]
@bartlomiejmarszal unit tests are failing

**Review by Karol-Stelmaczonek — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: test/unit/models/classes/Tool/Service/AuthoringLtiRoleServiceTest.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    {
+        return [
+            'When one valid roles' => [
+                'rolesProvided' => [
+                    'http://purl.imsglobal.org/vocab/lis/v2/institution/person#Administrator',
+                    'http://purl.imsglobal.org/vocab/lis/v2/membership/Administrator#Support'
+                ],
+                'expected' => 'http://purl.imsglobal.org/vocab/lis/v2/institution/person#Administrator'
+            ],
+            'When more then one valid roles' => [
```
Comment (pribi) [inline test/unit/models/classes/Tool/Service/AuthoringLtiRoleServiceTest.php:84]: typo: then => than

**Review by andreluizmachado — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`

**Review by Karol-Stelmaczonek — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**